packages/pipelines/src/types/k8s.ts

```diff
@@ -96,3 +99,196 @@ export type K8sResourceCommon = K8sResourceIdentifier &
+  crd?: boolean;
+  apiVersion: string;
+  apiGroup?: string;
+  namespaced?: boolean;
+  selector?: Selector;
+  labels?: { [key: string]: string };
+  annotations?: { [key: string]: string };
+  verbs?: K8sVerb[];
+  shortNames?: string[];
+  badge?: BadgeType;
+  color?: string;
+
+  // Legacy option for supporing plural names in URL paths when `crd: true`.
+  // This should not be set for new models, but is needed to avoid breaking
+  // existing links as we transition to using the API group in URL paths.
+  legacyPluralURL?: boolean;
+};
+
+export type QueryParams = {
+  watch?: string;
+  labelSelector?: string;
+  fieldSelector?: string;
+  resourceVersion?: string;
+  [key: string]: string | undefined;
+};
+
+export type Options = {
+  ns?: string;
+  name?: string;
+  path?: string;
+  queryParams?: QueryParams;
+  cluster?: string;
+};
+
+export type K8sGet = <R extends K8sResourceCommon>(options: {
+  model: K8sModel;
+  name: string;
+  ns?: string;
+  path?: string;
+  queryParams?: QueryParams;
+  requestInit?: RequestInit;
+}) => Promise<R>;
+
+export type WatchK8sResult<R extends K8sResourceCommon | K8sResourceCommon[]> = [R, boolean, any];
+
+export type UseK8sWatchResource = <R extends K8sResourceCommon | K8sResourceCommon[]>(
+  initResource: WatchK8sResource | null,
+) => WatchK8sResult<R>;
+
+declare type BaseOptions = {
+  name?: string;
+  ns?: string;
+  path?: string;
+  queryParams?: QueryParams;
+};
+declare type OptionsGet = BaseOptions & {
+  model: K8sModel;
+  requestInit?: RequestInit;
+};
+
+export type commonFetchJSON = {
+  <TResult>(
+    url: string,
+    requestInit?: RequestInit | undefined,
+    timeout?: number | undefined,
+    isK8sAPIRequest?: boolean | undefined,
+  ): Promise<TResult>;
+  put<TResult_1>(
+    url: string,
+    data: unknown,
+    requestInit?: RequestInit | undefined,
+    timeout?: number | undefined,
+    isK8sAPIRequest?: boolean | undefined,
+  ): Promise<TResult_1>;
+  post<TResult_2>(
+    url: string,
+    data: unknown,
+    requestInit?: RequestInit | undefined,
+    timeout?: number | undefined,
+    isK8sAPIRequest?: boolean | undefined,
+  ): Promise<TResult_2>;
+  patch<TResult_3>(
+    url: string,
+    data: unknown,
+    requestInit?: RequestInit | undefined,
+    timeout?: number | undefined,
+    isK8sAPIRequest?: boolean | undefined,
+  ): Promise<TResult_3>;
+  delete<TResult_4>(
+    url: string,
+    data?: unknown,
+    requestInit?: RequestInit | undefined,
+    timeout?: number | undefined,
+    isK8sAPIRequest?: boolean | undefined,
+  ): Promise<TResult_4>;
+};
+
+export type commonFetchText = (
+  url: string,
+  requestInit?: RequestInit | undefined,
+  timeout?: number | undefined,
+  isK8sAPIRequest?: boolean | undefined,
+) => Promise<string>;
+
+export type K8sGetResource = <R extends K8sResourceCommon>(options: OptionsGet) => Promise<R>;
+
+export type FetchUtilsType = {
+  hooks: { useK8sWatchResource: UseK8sWatchResource; k8sGet?: K8sGetResource };
```

Comment:
I think `k8sGet` is needed here to fetch the tekton results service URL, since we are going to pass the URL through config parameters, if this is not needed anymore then can we remove it?